onnxruntime/test/python/quantization/test_op_slice.py

```diff
@@ -0,0 +1,162 @@
+        ]
+        slice_input_names = ["input_0", "starts", "ends"]
+
+        if axes:
+            initializers.append(onnx.numpy_helper.from_array(np.array(axes, dtype=np.int64), "axes"))
+            slice_input_names.append("axes")
+
+        if steps:
+            if not axes:
+                slice_input_names.append("")  # Empty axes input.
+            initializers.append(onnx.numpy_helper.from_array(np.array(steps, dtype=np.int64), "steps"))
+            slice_input_names.append("steps")
+
+        slice_node = onnx.helper.make_node("Slice", slice_input_names, ["output_0"], name="Slice0")
+
+        graph = onnx.helper.make_graph(
+            [slice_node],
+            "SliceGraph",
+            [input_0],
+            [output_0],
+            initializer=initializers,
+        )
+        opset_imports = [onnx.helper.make_opsetid("", 21)]
+        model = onnx.helper.make_model(graph, opset_imports=opset_imports)
+        model = onnx.shape_inference.infer_shapes(model)
+        onnx.checker.check_model(model, True)
+        return model
+
+    def test_qdq_slice_qparams(self):
+        """
+        Test that QDQ Slice has equal scale/zero-point for its input and output.
+        """
+        test_configs = [onnx.TensorProto.FLOAT, onnx.TensorProto.FLOAT16]
+
+        for onnx_tensor_type in test_configs:
+            with self.subTest(onnx_tensor_type=onnx_tensor_type):
+                label = f"{onnx.TensorProto.DataType.Name(onnx_tensor_type)}"
+                float_model_path = os.path.join(self._tmp_dir_path, f"slice.{label}.onnx")
+                qdq_model_path = os.path.join(self._tmp_dir_path, f"slice.{label}.qdq.onnx")
+
+                input_shape = [2, 4]
+                float_model = self.build_slice_model(
+                    input_shape=input_shape,
+                    input_tensor_type=onnx_tensor_type,
+                    starts=[1, 0],
+                    ends=[2, 3],
+                    axes=None,
+                    steps=[1, 2],
+                )
+                onnx.save_model(float_model, float_model_path)
+
+                # Create a data reader
+                np_dtype = onnx.helper.tensor_dtype_to_np_dtype(onnx_tensor_type)
+                input_data_list = [
+                    {"input_0": np.array([[1.0, 2.0, 3.0, 4.0], [5.0, 6.0, 7.0, 8.0]], dtype=np_dtype)},
+                    {"input_0": np.array([[-1.0, -2.0, -3.0, -4.0], [-5.0, -6.0, -7.0, -8.0]], dtype=np_dtype)},
+                ]
+                data_reader = TestDataFeeds(input_data_list)
+
+                # quantize model to QDQ
+                quantize_static(
+                    float_model_path,
+                    qdq_model_path,
+                    data_reader,
+                    quant_format=QuantFormat.QDQ,
+                    activation_type=QuantType.QUInt8,
+                    weight_type=QuantType.QInt8,
+                    extra_options={"ForceQuantizeNoInputCheck": True},
```

Comment:
Note: This is needed to force quantization of Slice that consumes a model input. By default, the qdq quantizer will not quantize a Slice if it gets its input from model input. See:
https://github.com/microsoft/onnxruntime/blob/c4fb724e810bb496165b9015c77f402727392933/onnxruntime/python/tools/quantization/quantize.py#L360-L363